_examples/metrics/per_second_data_collector.go

```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"sync"
+	"time"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+)
+
+var (
+	perSecondDataCollector = operatormetrics.Collector{
+		Metrics: []operatormetrics.Metric{
+			perSecondData,
+		},
+		CollectCallback: perSecondDataCollectorCallback,
+	}
+
+	perSecondData = operatormetrics.NewGaugeVec(
+		operatormetrics.MetricOpts{
+			Name: metricPrefix + "per_second_data",
+			Help: "Data per second",
+		},
+		[]string{"source"},
+	)
+
+	queue = make([]data, 0)
```

Comment:
you can also use linked list `queue = list.New()` instead of  slice-based, something like this:

`
for node := queue.Front(); node != nil; node = node.Next() {
	item := node.Value.(data)
	crs = append(crs, operatormetrics.CollectorResult{
		Metric:    perSecondData,
		Labels:    []string{item.source},
		Value:     item.value,
		Timestamp: item.timestamp,
	})
	queue.Remove(node)
}
`

Reply:
i dont think its more efficient, i saw that kubernetes workqueue also uses slices for example